tools/depends/configure.ac

```diff
@@ -367,7 +367,7 @@ case $host in
         platform_os="ios"
 
         if [ ! test "x$use_cpu" = "xarm64" ]; then
-          platform_cflags+=" -mcpu=cortex-a8 -mfpu=neon"
+          platform_cflags+=" -mfpu=neon"
```

Comment:
removing `-mcpu=cortex-a8` is wrong, it is supported for armv7 builds

Reply:
xcode 10 errors with unknown mcpu.

Ill get a local build going to get the exact error, but if you can locate the previous jenkins build, it will have the error

Reply:
Found the log - https://jenkins.kodi.tv/job/IOS/10085/console

```
12:22:25     clang: error: the clang compiler does not support '-mcpu=cortex-a8'
12:22:25     
12:22:25     ** BUILD FAILED **
12:22:25     
12:22:25     
12:22:25     The following build commands failed:
12:22:25     	CompileC CMAKE_TRY_COMPILE.build/Debug-iphoneos/cmTC_32c45.build/Objects-normal/arm64/testCCompiler.o testCCompiler.c normal arm64 c com.apple.compilers.llvm.clang.1_0.compiler
12:22:25     (1 failure)
12:22:25     
12:22:25 
12:22:25   
12:22:25 
12:22:25   CMake will not be able to correctly generate this project.
12:22:25 Call Stack (most recent call first):
12:22:25   CMakeLists.txt:2 (project)
12:22:25 
12:22:25 
12:22:25 -- Configuring incomplete, errors occurred!
```

Reply:
You cut the important thing, the invoked command.

`/Applications/Xcode10.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/clang -x c -arch arm64 -fmessage-length=0 -fdiagnostics-show-note-include-stack -fmacro-backtrace-limit=0 -Wno-trigraphs -fpascal-strings -O0 -Wno-missing-field-initializers -Wno-missing-prototypes -Wno-return-type -Wno-missing-braces -Wparentheses -Wswitch -Wno-unused-function -Wno-unused-label -Wno-unused-parameter -Wno-unused-variable -Wunused-value -Wno-empty-body -Wno-uninitialized -Wno-unknown-pragmas -Wno-shadow -Wno-four-char-constants -Wno-conversion -Wno-constant-conversion -Wno-int-conversion -Wno-bool-conversion -Wno-enum-conversion -Wno-float-conversion -Wno-non-literal-null-conversion -Wno-objc-literal-conversion -Wshorten-64-to-32 -Wpointer-sign -Wno-newline-eof -DCMAKE_INTDIR=\"Debug-iphoneos\" -isysroot /Applications/Xcode10.2.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -fstrict-aliasing -Wdeprecated-declarations -miphoneos-version-min=12.2 -g -Wno-sign-conversion -Wno-infinite-recursion -Wno-comma -Wno-block-capture-autoreleasing -Wno-strict-prototypes -Wno-semicolon-before-method-body -I/Users/Shared/jenkins/workspace/IOS/build/CMakeFiles/CMakeTmp/Debug/include -I/Users/Shared/jenkins/workspace/IOS/build/CMakeFiles/CMakeTmp/CMAKE_TRY_COMPILE.build/Debug-iphoneos/cmTC_32c45.build/DerivedSources-normal/arm64 -I/Users/Shared/jenkins/workspace/IOS/build/CMakeFiles/CMakeTmp/CMAKE_TRY_COMPILE.build/Debug-iphoneos/cmTC_32c45.build/DerivedSources/arm64 -I/Users/Shared/jenkins/workspace/IOS/build/CMakeFiles/CMakeTmp/CMAKE_TRY_COMPILE.build/Debug-iphoneos/cmTC_32c45.build/DerivedSources -Wmost -Wno-four-char-constants -Wno-unknown-pragmas -F/Users/Shared/jenkins/workspace/IOS/build/CMakeFiles/CMakeTmp/Debug -fheinous-gnu-extensions -no-cpp-precomp -mcpu=cortex-a8 -mfpu=neon -ftree-vectorize -pipe -Wno-trigraphs -fpascal-strings -Wreturn-type -Wunused-variable -fmessage-length=0 -Wno-error=implicit-function-declaration -arch armv7 -miphoneos-version-min=9.0 -D_DEBUG -isysroot /Applications/Xcode10.2.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -isystem /Users/Shared/jenkins/workspace/IOS/tools/depends/xbmc-depends/iphoneos12.2_armv7-target-debug/include -fheinous-gnu-extensions -no-cpp-precomp -mcpu=cortex-a8 -mfpu=neon -ftree-vectorize -pipe -Wno-trigraphs -fpascal-strings -Wreturn-type -Wunused-variable -fmessage-length=0 -Wno-error=implicit-function-declaration -arch armv7 -miphoneos-version-min=9.0 -D_DEBUG -isysroot /Applications/Xcode10.2.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -isystem /Users/Shared/jenkins/workspace/IOS/tools/depends/xbmc-depends/iphoneos12.2_armv7-target-debug/include -MMD -MT dependencies -MF /Users/Shared/jenkins/workspace/IOS/build/CMakeFiles/CMakeTmp/CMAKE_TRY_COMPILE.build/Debug-iphoneos/cmTC_32c45.build/Objects-normal/arm64/testCCompiler.d --serialize-diagnostics /Users/Shared/jenkins/workspace/IOS/build/CMakeFiles/CMakeTmp/CMAKE_TRY_COMPILE.build/Debug-iphoneos/cmTC_32c45.build/Objects-normal/arm64/testCCompiler.dia -c /Users/Shared/jenkins/workspace/IOS/build/CMakeFiles/CMakeTmp/testCCompiler.c -o /Users/Shared/jenkins/workspace/IOS/build/CMakeFiles/CMakeTmp/CMAKE_TRY_COMPILE.build/Debug-iphoneos/cmTC_32c45.build/Objects-normal/arm64/testCCompiler.o`

You see the wrong argument?

Reply:
ive no idea how/where it switches to arm64 for that test though. The below is what the configure command sets in that build failure.

```
11:06:24 #------- configuration -------#
11:06:24 -e ccache:		 yes
11:06:24 -e build type:	 debug
11:06:24 -e toolchain:	 /Applications/Xcode10.2.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain
11:06:24 -e cpu:		 armv7
11:06:24 -e host:		 arm-apple-darwin
11:06:24 -e cflags:		 -fheinous-gnu-extensions -no-cpp-precomp -mcpu=cortex-a8 -mfpu=neon -ftree-vectorize -pipe -Wno-trigraphs -fpascal-strings -Wreturn-type -Wunused-variable -fmessage-length=0 -gdwarf-2 -Wno-error=implicit-function-declaration -arch armv7 -miphoneos-version-min=9.0 -g -D_DEBUG 
11:06:24 -e cxxflags:	 -std=c++11 -no-cpp-precomp -arch armv7 -miphoneos-version-min=9.0 -stdlib=libc++ -g -D_DEBUG 
11:06:24 -e ldflags:	 -Wl,-search_paths_first -Wl,-segalign,4000 -L/Applications/Xcode10.2.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk/usr/lib/system -arch armv7 -miphoneos-version-min=9.0 -isysroot /Applications/Xcode10.2.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS12.2.sdk -stdlib=libc++  -liconv 
11:06:24 -e ffmpeg options:	 
11:06:24 -e prefix:		 /Users/Shared/jenkins/workspace/IOS/tools/depends/xbmc-depends
11:06:24 -e depends:	 /Users/Shared/jenkins/workspace/IOS/tools/depends/xbmc-depends/iphoneos12.2_armv7-target-debug
```

Reply:
cmakeoutput.log

https://paste.kodi.tv/ukecevakuj

cmakeerror.log

https://paste.kodi.tv/azaseyajax

Reply:
I invoked the command to generated the Xcode project for armv7 and after it failed searched for arm64 in the build folder.
```
$ grep --color -R arm64 *
CMakeFiles/3.12.2/CMakeCXXCompiler.cmake:set(CMAKE_XCODE_ARCHS "arm64")
[...]
```

Reply:
yeah, looking at that right now. going down the path of a cmake issue? i really dont know at this point sorry. doing some searching now.

Reply:
can add the following to toolchain.cmake, and it forces it to skip compiler tests. i know thats a heavy handed way around it, but just documenting another method for now.

```
set(CMAKE_C_COMPILER_FORCED TRUE)
```

no need to force CMAKE_CXX_COMPILER, as it either doest test, or doesnt fail the test.

Reply:
even with that (keeping the -mcpu=cortex-a8 flag), libfmt still fails with the same information provided above.

Reply:
That's because `set(CMAKE_C_COMPILER_FORCED TRUE)` is a hack. You have to tell cmake to configure for armv7. Do not try to tell cmake to skip compiler checks.

Reply:
as far as i can see, all the things i can find are saying its already been told to build for armv7. Ive no idea why that test does what it does.

Kambala has confirmed both the cmake test hack and the cortex-a8 removal and the libfmt fix end in runnable builds on ios9 devices.

Honestly, i see 32bit ios as a niche. I know you might not see it that way, but 32bit in the apple world is/has gone the way of the dodo. unless someone can point me in the right direction for an acceptable solution, i personally dont think the time invested so far for it is worth it.

Reply:
No, otherwise the compiler check wouldn't pass `-arch arm64` as argument.

I never said it doesn't work, only that the this is a hack and therefore not acceptable.

If you think that's the only way, okay. In that case you have to do a complete removal, only stop building for ios armv7 it isn't enough.

Reply:
So I tend to drop armv7 support for iOS then.

Reply:
No longer needed, thanks for the fix Rechi